### PR TITLE
Adds Universal Link Support 

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

1. Setups up the API to serve
 - An Apple specific manifest
  - A Google specific manifest 
  - A redirect to the Play/Apple store when visiting specific subroutes of the API.

2. Sets up Android/iOS to look for said manifests and then open them as app links. 

3. Overrides some features on the NotificationsProvider to scrub out the needed parts of the path to properly open deep links.

4. Adds support to the navigation service for navigating without errors if the navigator hasn't been loaded yet.

### What design trade-offs/decisions were made?

Tried to package things up as best as possible so in the future we can bring this functionality to corre. 

#### iOS

1. Text yourself the following code
https://willowcreek-api-staging.herokuapp.com/app-link/ContentSingle?itemId=ContentSeriesContentItem:8a8c380f91bfabe799fde880a222b2a5 - you should end up on the app store
2. Build the app for your phone, doesn't matter if you do a dev/prod build.
3. Tap the link again, you should be redirected to a piece of content in the app

#### Android

1. Open an emulator
2. Delete the Willow app from the emulator.
3. Run this command in your shell: 
```
adb shell am start -a android.intent.action.VIEW \
    -c android.intent.category.BROWSABLE \
    -d "https://willowcreek-api-staging.herokuapp.com/app-link/ContentSingle\?itemId\=ContentSeriesContentItem:8a8c380f91bfabe799fde880a222b2a5"
```
4. You should be redirected to the Play Store
5. Build the app for your android emulator
6. Run the command above again, you should end up at the content in the willow app.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
image

**Comment by vinnyjth** [comment]
OH! Make sure you turn off storybook to test this. Storybook makes it very difficult to test this properly because the Navigator isn't mounted when you load the default storybook screen.

File: packages/WillowCreekApp/index.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ import './loadConfig';
 import './src/bugsnag';
 
 import { AppRegistry, YellowBox } from 'react-native';
-import SplashScreen from 'react-native-splash-screen';
```
[the remark this answers is not in the document]
Comment (vinnyjth) [reply packages/WillowCreekApp/index.js]: I'm not sure I understand the question.
